runtime/vm/dart_api_impl.cc

```diff
@@ -1286,7 +1286,7 @@ static Dart_Isolate CreateIsolate(IsolateGroup* group,
     if (error != NULL) {
       *error = Utils::StrDup("Isolate creation failed");
     }
-    return reinterpret_cast<Dart_Isolate>(NULL);
+    return reinterpret_cast<Dart_Isolate>(0);
```

Comment:
```
FAILED: obj/runtime/vm/libdart_jit.dart_api_impl.o 
/usr/bin/clang++ -MMD -MF obj/runtime/vm/libdart_jit.dart_api_impl.o.d -DTOOLCHAIN_VERSION=aaf7SYkyfNEKVhRYxwHEgw9xHEnquJ9bO3avIbhc2qoC -DNDEBUG -DTARGET_ARCH_X64 -DDART_TARGET_OS_LINUX -DDART_SHARED_LIB -I../../runtime -I../.. -Igen -I../../runtime/include -m64 -march=x86-64 -msse2 -fPIC -fcolor-diagnostics -Wall -Wextra -Wendif-labels -Wno-missing-field-initializers -Wno-unused-parameter -Wno-tautological-constant-compare -Wno-unused-but-set-variable -Wno-unused-but-set-parameter -Wno-deprecated-non-prototype -fdebug-prefix-map=/home/runner/work/test/test/dart-sdk/sdk=../.. -no-canonical-prefixes --sysroot=../../build/linux/alpine-linux-x86_64-sysroot -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -Wheader-hygiene -Wstring-conversion -O3 -fno-ident -fdata-sections -ffunction-sections -g3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-unused-private-field -Wnon-virtual-dtor -Wvla -Woverloaded-virtual -Wno-comments -g3 -ggdb3 -fno-rtti -fno-exceptions -Wimplicit-fallthrough -fno-strict-vtable-pointers -O3 -fvisibility-inlines-hidden -fno-omit-frame-pointer -std=c++17 -fno-rtti -c ../../runtime/vm/dart_api_impl.cc -o obj/runtime/vm/libdart_jit.dart_api_impl.o
../../runtime/vm/dart_api_impl.cc:1289:12: error: reinterpret_cast from 'std::nullptr_t' to 'Dart_Isolate' (aka '_Dart_Isolate *') is not allowed
    return reinterpret_cast<Dart_Isolate>(NULL);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../runtime/vm/dart_api_impl.cc:1333:10: error: reinterpret_cast from 'std::nullptr_t' to 'Dart_Isolate' (aka '_Dart_Isolate *') is not allowed
  return reinterpret_cast<Dart_Isolate>(NULL);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

I found this on musl libc's mail list: https://www.openwall.com/lists/musl/2021/12/24/1
May be this should be `static_cast`?

Reply:
The issue is that glibc defines `NULL` as `__null` but musl defines `NULL` to `0` in C++, which works differently.

Reply:
https://en.cppreference.com/w/cpp/language/reinterpret_cast

> 3) A value of any integral or enumeration type can be converted to a pointer type. A pointer converted to an integer of sufficient size and back to the same pointer type is guaranteed to have its original value, otherwise the resulting pointer cannot be dereferenced safely (the round-trip conversion in the opposite direction is not guaranteed; the same pointer may have multiple integer representations) The null pointer constant [NULL](https://en.cppreference.com/w/cpp/types/NULL) or integer zero is not guaranteed to yield the null pointer value of the target type; [static_cast](https://en.cppreference.com/w/cpp/language/static_cast) or [implicit conversion](https://en.cppreference.com/w/cpp/language/implicit_cast) should be used for this purpose.

Reply:
@mraleph Based on the c++ documentation I have replaced all `[reinterpret_cast<T*>(NULL)` with `static_cast<T*>(NULL)`. I think this should be a more appropriate fix.